instrumentation/opentelemetry-instrumentation-pyramid/src/opentelemetry/instrumentation/pyramid/__init__.py

```diff
@@ -94,9 +92,12 @@
     trace_tween_factory,
 )
 from opentelemetry.instrumentation.pyramid.package import _instruments
-from opentelemetry.instrumentation.pyramid.version import __version__
 from opentelemetry.instrumentation.utils import unwrap
-from opentelemetry.trace import TracerProvider, get_tracer
+
+# test_automatic.TestAutomatic.test_tween_list needs trace_tween_factory to be
+# imported in this module. The next line is necessary to avoid a lint error
+# from importing an unused symbol.
```

Comment:
I was trying to look a the test but it's not immediately obvious to me how it uses `test_tween_list`, is it trying to mock the value?

https://github.com/open-telemetry/opentelemetry-python-contrib/blob/cc57aacd71c3eff8fac4030ca3013546dd4e545a/instrumentation/opentelemetry-instrumentation-pyramid/tests/test_automatic.py#L59-L67

It would be nice to see if there was a way that we can remove this unused import so that the test doesn't drive the imports here, but I think it's fair to say it's out of scope for this PRs change.

Reply:
Just for the record, here is the error I got:

```
py39-test-instrumentation-pyramid installed: asgiref==3.4.1,attrs==21.2.0,click==7.1.2,Deprecated==1.2.12,Flask==1.1.4,hupper==1.10.3,iniconfig==1.1.1,itsdangerous==1.1.0,Jinja2==2.11.3,MarkupSafe==2.0.1,opentelemetry-api @ file:///home/ocelotl/github/ocelotl/opentelemetry-python-contrib/opentelemetry-python-core/opentelemetry-api,opentelemetry-instrumentation @ file:///home/ocelotl/github/ocelotl/opentelemetry-python-contrib/opentelemetry-python-core/opentelemetry-instrumentation,opentelemetry-instrumentation-pyramid @ file:///home/ocelotl/github/ocelotl/opentelemetry-python-contrib/instrumentation/opentelemetry-instrumentation-pyramid,opentelemetry-instrumentation-wsgi @ file:///home/ocelotl/github/ocelotl/opentelemetry-python-contrib/instrumentation/opentelemetry-instrumentation-wsgi,opentelemetry-sdk @ file:///home/ocelotl/github/ocelotl/opentelemetry-python-contrib/opentelemetry-python-core/opentelemetry-sdk,opentelemetry-semantic-conventions @ file:///home/ocelotl/github/ocelotl/opentelemetry-python-contrib/opentelemetry-python-core/opentelemetry-semantic-conventions,opentelemetry-test @ file:///home/ocelotl/github/ocelotl/opentelemetry-python-contrib/opentelemetry-python-core/tests/util,opentelemetry-util-http @ file:///home/ocelotl/github/ocelotl/opentelemetry-python-contrib/util/opentelemetry-util-http,packaging==21.0,PasteDeploy==2.1.1,plaster==1.0,plaster-pastedeploy==0.7,pluggy==1.0.0,py==1.10.0,py-cpuinfo==8.0.0,pyparsing==2.4.7,pyramid==2.0,pytest==6.2.5,pytest-benchmark==3.4.1,toml==0.10.2,translationstring==1.4,venusian==3.0.0,WebOb==1.8.7,Werkzeug==0.16.1,wrapt==1.12.1,zope.deprecation==4.4.0,zope.interface==5.4.0
py39-test-instrumentation-pyramid run-test-pre: PYTHONHASHSEED='2266245183'
py39-test-instrumentation-pyramid run-test-pre: commands[0] | python -m pip install -U pip setuptools wheel
Requirement already satisfied: pip in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (21.2.4)
Requirement already satisfied: setuptools in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (57.4.0)
Requirement already satisfied: wheel in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (0.37.0)
py39-test-instrumentation-pyramid run-test-pre: commands[1] | pip install '/home/ocelotl/github/ocelotl/opentelemetry-python-contrib/opentelemetry-python-core/opentelemetry-api[test]'
Processing /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/opentelemetry-python-core/opentelemetry-api
  DEPRECATION: A future pip version will change local packages to be built in-place without first copying to a temporary directory. We recommend you use --use-feature=in-tree-build to test your packages with this new behavior before it becomes the default.
   pip 21.3 will remove support for this functionality. You can find discussion regarding this at https://github.com/pypa/pip/issues/7555.
Requirement already satisfied: Deprecated>=1.2.6 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from opentelemetry-api==1.5.0) (1.2.12)
Requirement already satisfied: wrapt<2,>=1.10 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from Deprecated>=1.2.6->opentelemetry-api==1.5.0) (1.12.1)
Building wheels for collected packages: opentelemetry-api
  Building wheel for opentelemetry-api (setup.py): started
  Building wheel for opentelemetry-api (setup.py): finished with status 'done'
  Created wheel for opentelemetry-api: filename=opentelemetry_api-1.5.0-py3-none-any.whl size=38435 sha256=9d05c5894a53802fac50cd1cdc9cf5e338b6859160a822fd2ae77bc4036fce63
  Stored in directory: /home/ocelotl/.cache/pip/wheels/dc/fe/0c/f4056fb463eceb4659daee29b5b64bd63bdce7047f8daf0dd4
Successfully built opentelemetry-api
Installing collected packages: opentelemetry-api
  Attempting uninstall: opentelemetry-api
    Found existing installation: opentelemetry-api 1.5.0
    Uninstalling opentelemetry-api-1.5.0:
      Successfully uninstalled opentelemetry-api-1.5.0
Successfully installed opentelemetry-api-1.5.0
py39-test-instrumentation-pyramid run-test-pre: commands[2] | pip install '/home/ocelotl/github/ocelotl/opentelemetry-python-contrib/opentelemetry-python-core/opentelemetry-semantic-conventions[test]'
Processing /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/opentelemetry-python-core/opentelemetry-semantic-conventions
  DEPRECATION: A future pip version will change local packages to be built in-place without first copying to a temporary directory. We recommend you use --use-feature=in-tree-build to test your packages with this new behavior before it becomes the default.
   pip 21.3 will remove support for this functionality. You can find discussion regarding this at https://github.com/pypa/pip/issues/7555.
Building wheels for collected packages: opentelemetry-semantic-conventions
  Building wheel for opentelemetry-semantic-conventions (setup.py): started
  Building wheel for opentelemetry-semantic-conventions (setup.py): finished with status 'done'
  Created wheel for opentelemetry-semantic-conventions: filename=opentelemetry_semantic_conventions-0.24b0-py3-none-any.whl size=23206 sha256=0cdc220003f32b8f905c54f98e9286b0542136796fd59abc5f66dc5f47d4074a
  Stored in directory: /home/ocelotl/.cache/pip/wheels/51/28/81/795111938ff19974deec501a21152c84636ea8dacd3dfcf162
Successfully built opentelemetry-semantic-conventions
Installing collected packages: opentelemetry-semantic-conventions
  Attempting uninstall: opentelemetry-semantic-conventions
    Found existing installation: opentelemetry-semantic-conventions 0.24b0
    Uninstalling opentelemetry-semantic-conventions-0.24b0:
      Successfully uninstalled opentelemetry-semantic-conventions-0.24b0
Successfully installed opentelemetry-semantic-conventions-0.24b0
py39-test-instrumentation-pyramid run-test-pre: commands[3] | pip install '/home/ocelotl/github/ocelotl/opentelemetry-python-contrib/opentelemetry-python-core/opentelemetry-instrumentation[test]'
Processing /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/opentelemetry-python-core/opentelemetry-instrumentation
  DEPRECATION: A future pip version will change local packages to be built in-place without first copying to a temporary directory. We recommend you use --use-feature=in-tree-build to test your packages with this new behavior before it becomes the default.
   pip 21.3 will remove support for this functionality. You can find discussion regarding this at https://github.com/pypa/pip/issues/7555.
Requirement already satisfied: opentelemetry-api~=1.4 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from opentelemetry-instrumentation==0.24b0) (1.5.0)
Requirement already satisfied: wrapt<2.0.0,>=1.0.0 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from opentelemetry-instrumentation==0.24b0) (1.12.1)
Requirement already satisfied: Deprecated>=1.2.6 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from opentelemetry-api~=1.4->opentelemetry-instrumentation==0.24b0) (1.2.12)
Building wheels for collected packages: opentelemetry-instrumentation
  Building wheel for opentelemetry-instrumentation (setup.py): started
  Building wheel for opentelemetry-instrumentation (setup.py): finished with status 'done'
  Created wheel for opentelemetry-instrumentation: filename=opentelemetry_instrumentation-0.24b0-py3-none-any.whl size=22612 sha256=4a243686b7f1a4a46fabe31de5739d38b1d598202620a10a9e19fd9ce630ff0d
  Stored in directory: /home/ocelotl/.cache/pip/wheels/3d/d3/52/0bcabea0c69432b5f5467c53ba45e1e2a17dadba58696e392c
Successfully built opentelemetry-instrumentation
Installing collected packages: opentelemetry-instrumentation
  Attempting uninstall: opentelemetry-instrumentation
    Found existing installation: opentelemetry-instrumentation 0.24b0
    Uninstalling opentelemetry-instrumentation-0.24b0:
      Successfully uninstalled opentelemetry-instrumentation-0.24b0
Successfully installed opentelemetry-instrumentation-0.24b0
py39-test-instrumentation-pyramid run-test-pre: commands[4] | pip install '/home/ocelotl/github/ocelotl/opentelemetry-python-contrib/opentelemetry-python-core/opentelemetry-sdk[test]'
Processing /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/opentelemetry-python-core/opentelemetry-sdk
  DEPRECATION: A future pip version will change local packages to be built in-place without first copying to a temporary directory. We recommend you use --use-feature=in-tree-build to test your packages with this new behavior before it becomes the default.
   pip 21.3 will remove support for this functionality. You can find discussion regarding this at https://github.com/pypa/pip/issues/7555.
Requirement already satisfied: opentelemetry-api==1.5.0 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from opentelemetry-sdk==1.5.0) (1.5.0)
Requirement already satisfied: opentelemetry-semantic-conventions==0.24b0 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from opentelemetry-sdk==1.5.0) (0.24b0)
Requirement already satisfied: opentelemetry-instrumentation==0.24b0 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from opentelemetry-sdk==1.5.0) (0.24b0)
Requirement already satisfied: Deprecated>=1.2.6 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from opentelemetry-api==1.5.0->opentelemetry-sdk==1.5.0) (1.2.12)
Requirement already satisfied: wrapt<2.0.0,>=1.0.0 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from opentelemetry-instrumentation==0.24b0->opentelemetry-sdk==1.5.0) (1.12.1)
Building wheels for collected packages: opentelemetry-sdk
  Building wheel for opentelemetry-sdk (setup.py): started
  Building wheel for opentelemetry-sdk (setup.py): finished with status 'done'
  Created wheel for opentelemetry-sdk: filename=opentelemetry_sdk-1.5.0-py3-none-any.whl size=41018 sha256=2039b91c9f20e3f32d28f57b25b9ffc261ab5a4ec7c78a3a0cea66d011040d37
  Stored in directory: /home/ocelotl/.cache/pip/wheels/bd/d0/d3/8c4a340bf050b99090576b0aa1d409920cbe8e5135116a6df9
Successfully built opentelemetry-sdk
Installing collected packages: opentelemetry-sdk
  Attempting uninstall: opentelemetry-sdk
    Found existing installation: opentelemetry-sdk 1.5.0
    Uninstalling opentelemetry-sdk-1.5.0:
      Successfully uninstalled opentelemetry-sdk-1.5.0
Successfully installed opentelemetry-sdk-1.5.0
py39-test-instrumentation-pyramid run-test-pre: commands[5] | pip install '/home/ocelotl/github/ocelotl/opentelemetry-python-contrib/opentelemetry-python-core/tests/util[test]'
Processing /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/opentelemetry-python-core/tests/util
  DEPRECATION: A future pip version will change local packages to be built in-place without first copying to a temporary directory. We recommend you use --use-feature=in-tree-build to test your packages with this new behavior before it becomes the default.
   pip 21.3 will remove support for this functionality. You can find discussion regarding this at https://github.com/pypa/pip/issues/7555.
Requirement already satisfied: opentelemetry-api~=1.3 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from opentelemetry-test==0.24b0) (1.5.0)
Requirement already satisfied: opentelemetry-sdk~=1.3 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from opentelemetry-test==0.24b0) (1.5.0)
Requirement already satisfied: flask~=1.0 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from opentelemetry-test==0.24b0) (1.1.4)
Requirement already satisfied: itsdangerous<2.0,>=0.24 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from flask~=1.0->opentelemetry-test==0.24b0) (1.1.0)
Requirement already satisfied: Jinja2<3.0,>=2.10.1 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from flask~=1.0->opentelemetry-test==0.24b0) (2.11.3)
Requirement already satisfied: Werkzeug<2.0,>=0.15 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from flask~=1.0->opentelemetry-test==0.24b0) (0.16.1)
Requirement already satisfied: click<8.0,>=5.1 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from flask~=1.0->opentelemetry-test==0.24b0) (7.1.2)
Requirement already satisfied: MarkupSafe>=0.23 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from Jinja2<3.0,>=2.10.1->flask~=1.0->opentelemetry-test==0.24b0) (2.0.1)
Requirement already satisfied: Deprecated>=1.2.6 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from opentelemetry-api~=1.3->opentelemetry-test==0.24b0) (1.2.12)
Requirement already satisfied: wrapt<2,>=1.10 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from Deprecated>=1.2.6->opentelemetry-api~=1.3->opentelemetry-test==0.24b0) (1.12.1)
Requirement already satisfied: opentelemetry-instrumentation==0.24b0 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from opentelemetry-sdk~=1.3->opentelemetry-test==0.24b0) (0.24b0)
Requirement already satisfied: opentelemetry-semantic-conventions==0.24b0 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from opentelemetry-sdk~=1.3->opentelemetry-test==0.24b0) (0.24b0)
Building wheels for collected packages: opentelemetry-test
  Building wheel for opentelemetry-test (setup.py): started
  Building wheel for opentelemetry-test (setup.py): finished with status 'done'
  Created wheel for opentelemetry-test: filename=opentelemetry_test-0.24b0-py3-none-any.whl size=7561 sha256=9aa841a800d30df32bf42e0df5e3c067311452b1ba1026f257f2d9134dbc8198
  Stored in directory: /tmp/pip-ephem-wheel-cache-50f8rwl0/wheels/9d/19/79/f059b55adcdc5f58cea759dcfbbc83d4188bd2867eee66646e
Successfully built opentelemetry-test
Installing collected packages: opentelemetry-test
  Attempting uninstall: opentelemetry-test
    Found existing installation: opentelemetry-test 0.24b0
    Uninstalling opentelemetry-test-0.24b0:
      Successfully uninstalled opentelemetry-test-0.24b0
Successfully installed opentelemetry-test-0.24b0
py39-test-instrumentation-pyramid run-test-pre: commands[6] | pip install '/home/ocelotl/github/ocelotl/opentelemetry-python-contrib/util/opentelemetry-util-http[test]'
Processing /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/util/opentelemetry-util-http
  DEPRECATION: A future pip version will change local packages to be built in-place without first copying to a temporary directory. We recommend you use --use-feature=in-tree-build to test your packages with this new behavior before it becomes the default.
   pip 21.3 will remove support for this functionality. You can find discussion regarding this at https://github.com/pypa/pip/issues/7555.
Requirement already satisfied: opentelemetry-api~=1.3 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from opentelemetry-util-http==0.24b0) (1.5.0)
Requirement already satisfied: opentelemetry-instrumentation==0.24b0 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from opentelemetry-util-http==0.24b0) (0.24b0)
Requirement already satisfied: asgiref~=3.0 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from opentelemetry-util-http==0.24b0) (3.4.1)
Requirement already satisfied: opentelemetry-test==0.24b0 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from opentelemetry-util-http==0.24b0) (0.24b0)
Requirement already satisfied: wrapt<2.0.0,>=1.0.0 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from opentelemetry-instrumentation==0.24b0->opentelemetry-util-http==0.24b0) (1.12.1)
Requirement already satisfied: opentelemetry-sdk~=1.3 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from opentelemetry-test==0.24b0->opentelemetry-util-http==0.24b0) (1.5.0)
Requirement already satisfied: Deprecated>=1.2.6 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from opentelemetry-api~=1.3->opentelemetry-util-http==0.24b0) (1.2.12)
Requirement already satisfied: opentelemetry-semantic-conventions==0.24b0 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from opentelemetry-sdk~=1.3->opentelemetry-test==0.24b0->opentelemetry-util-http==0.24b0) (0.24b0)
Building wheels for collected packages: opentelemetry-util-http
  Building wheel for opentelemetry-util-http (setup.py): started
  Building wheel for opentelemetry-util-http (setup.py): finished with status 'done'
  Created wheel for opentelemetry-util-http: filename=opentelemetry_util_http-0.24b0-py3-none-any.whl size=3981 sha256=5a76c6e17ade0ee6873b75f6de015d591d87112affb82bc36c9559d706b352b8
  Stored in directory: /home/ocelotl/.cache/pip/wheels/d0/a4/28/5e084a3f421b7ef712e9d0eacfe9dedb3c30dece249072deda
Successfully built opentelemetry-util-http
Installing collected packages: opentelemetry-util-http
  Attempting uninstall: opentelemetry-util-http
    Found existing installation: opentelemetry-util-http 0.24b0
    Uninstalling opentelemetry-util-http-0.24b0:
      Successfully uninstalled opentelemetry-util-http-0.24b0
Successfully installed opentelemetry-util-http-0.24b0
py39-test-instrumentation-pyramid run-test-pre: commands[7] | pip install '/home/ocelotl/github/ocelotl/opentelemetry-python-contrib/instrumentation/opentelemetry-instrumentation-wsgi[test]'
Processing /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/instrumentation/opentelemetry-instrumentation-wsgi
  DEPRECATION: A future pip version will change local packages to be built in-place without first copying to a temporary directory. We recommend you use --use-feature=in-tree-build to test your packages with this new behavior before it becomes the default.
   pip 21.3 will remove support for this functionality. You can find discussion regarding this at https://github.com/pypa/pip/issues/7555.
Requirement already satisfied: opentelemetry-api~=1.3 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from opentelemetry-instrumentation-wsgi==0.24b0) (1.5.0)
Requirement already satisfied: opentelemetry-semantic-conventions==0.24b0 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from opentelemetry-instrumentation-wsgi==0.24b0) (0.24b0)
Requirement already satisfied: opentelemetry-instrumentation==0.24b0 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from opentelemetry-instrumentation-wsgi==0.24b0) (0.24b0)
Requirement already satisfied: opentelemetry-util-http==0.24b0 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from opentelemetry-instrumentation-wsgi==0.24b0) (0.24b0)
Requirement already satisfied: opentelemetry-test==0.24b0 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from opentelemetry-instrumentation-wsgi==0.24b0) (0.24b0)
Requirement already satisfied: wrapt<2.0.0,>=1.0.0 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from opentelemetry-instrumentation==0.24b0->opentelemetry-instrumentation-wsgi==0.24b0) (1.12.1)
Requirement already satisfied: opentelemetry-sdk~=1.3 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from opentelemetry-test==0.24b0->opentelemetry-instrumentation-wsgi==0.24b0) (1.5.0)
Requirement already satisfied: asgiref~=3.0 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from opentelemetry-util-http==0.24b0->opentelemetry-instrumentation-wsgi==0.24b0) (3.4.1)
Requirement already satisfied: Deprecated>=1.2.6 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from opentelemetry-api~=1.3->opentelemetry-instrumentation-wsgi==0.24b0) (1.2.12)
Building wheels for collected packages: opentelemetry-instrumentation-wsgi
  Building wheel for opentelemetry-instrumentation-wsgi (setup.py): started
  Building wheel for opentelemetry-instrumentation-wsgi (setup.py): finished with status 'done'
  Created wheel for opentelemetry-instrumentation-wsgi: filename=opentelemetry_instrumentation_wsgi-0.24b0-py3-none-any.whl size=10850 sha256=51c659543e3b31e7c0af2dc7db20342e2b3ff753081d6bd2e2f60d37d602cdfe
  Stored in directory: /home/ocelotl/.cache/pip/wheels/b3/00/4f/54cb9b3d57548f35e153da2c8f3b917d65c5c5a76cf5ac4a10
Successfully built opentelemetry-instrumentation-wsgi
Installing collected packages: opentelemetry-instrumentation-wsgi
  Attempting uninstall: opentelemetry-instrumentation-wsgi
    Found existing installation: opentelemetry-instrumentation-wsgi 0.24b0
    Uninstalling opentelemetry-instrumentation-wsgi-0.24b0:
      Successfully uninstalled opentelemetry-instrumentation-wsgi-0.24b0
Successfully installed opentelemetry-instrumentation-wsgi-0.24b0
py39-test-instrumentation-pyramid run-test-pre: commands[8] | pip install '/home/ocelotl/github/ocelotl/opentelemetry-python-contrib/instrumentation/opentelemetry-instrumentation-pyramid[test]'
Processing /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/instrumentation/opentelemetry-instrumentation-pyramid
  DEPRECATION: A future pip version will change local packages to be built in-place without first copying to a temporary directory. We recommend you use --use-feature=in-tree-build to test your packages with this new behavior before it becomes the default.
   pip 21.3 will remove support for this functionality. You can find discussion regarding this at https://github.com/pypa/pip/issues/7555.
Requirement already satisfied: opentelemetry-instrumentation==0.24b0 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from opentelemetry-instrumentation-pyramid==0.24b0) (0.24b0)
Requirement already satisfied: opentelemetry-api~=1.3 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from opentelemetry-instrumentation-pyramid==0.24b0) (1.5.0)
Requirement already satisfied: opentelemetry-semantic-conventions==0.24b0 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from opentelemetry-instrumentation-pyramid==0.24b0) (0.24b0)
Requirement already satisfied: opentelemetry-instrumentation-wsgi==0.24b0 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from opentelemetry-instrumentation-pyramid==0.24b0) (0.24b0)
Requirement already satisfied: opentelemetry-util-http==0.24b0 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from opentelemetry-instrumentation-pyramid==0.24b0) (0.24b0)
Requirement already satisfied: wrapt<2.0.0,>=1.0.0 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from opentelemetry-instrumentation-pyramid==0.24b0) (1.12.1)
Requirement already satisfied: werkzeug==0.16.1 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from opentelemetry-instrumentation-pyramid==0.24b0) (0.16.1)
Requirement already satisfied: opentelemetry-test==0.24b0 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from opentelemetry-instrumentation-pyramid==0.24b0) (0.24b0)
Requirement already satisfied: pyramid>=1.7 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from opentelemetry-instrumentation-pyramid==0.24b0) (2.0)
Requirement already satisfied: opentelemetry-sdk~=1.3 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from opentelemetry-test==0.24b0->opentelemetry-instrumentation-pyramid==0.24b0) (1.5.0)
Requirement already satisfied: asgiref~=3.0 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from opentelemetry-util-http==0.24b0->opentelemetry-instrumentation-pyramid==0.24b0) (3.4.1)
Requirement already satisfied: Deprecated>=1.2.6 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from opentelemetry-api~=1.3->opentelemetry-instrumentation-pyramid==0.24b0) (1.2.12)
Requirement already satisfied: webob>=1.8.3 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from pyramid>=1.7->opentelemetry-instrumentation-pyramid==0.24b0) (1.8.7)
Requirement already satisfied: plaster-pastedeploy in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from pyramid>=1.7->opentelemetry-instrumentation-pyramid==0.24b0) (0.7)
Requirement already satisfied: zope.deprecation>=3.5.0 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from pyramid>=1.7->opentelemetry-instrumentation-pyramid==0.24b0) (4.4.0)
Requirement already satisfied: plaster in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from pyramid>=1.7->opentelemetry-instrumentation-pyramid==0.24b0) (1.0)
Requirement already satisfied: venusian>=1.0 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from pyramid>=1.7->opentelemetry-instrumentation-pyramid==0.24b0) (3.0.0)
Requirement already satisfied: translationstring>=0.4 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from pyramid>=1.7->opentelemetry-instrumentation-pyramid==0.24b0) (1.4)
Requirement already satisfied: hupper>=1.5 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from pyramid>=1.7->opentelemetry-instrumentation-pyramid==0.24b0) (1.10.3)
Requirement already satisfied: setuptools in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from pyramid>=1.7->opentelemetry-instrumentation-pyramid==0.24b0) (57.4.0)
Requirement already satisfied: zope.interface>=3.8.0 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from pyramid>=1.7->opentelemetry-instrumentation-pyramid==0.24b0) (5.4.0)
Requirement already satisfied: PasteDeploy>=2.0 in /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages (from plaster-pastedeploy->pyramid>=1.7->opentelemetry-instrumentation-pyramid==0.24b0) (2.1.1)
Building wheels for collected packages: opentelemetry-instrumentation-pyramid
  Building wheel for opentelemetry-instrumentation-pyramid (setup.py): started
  Building wheel for opentelemetry-instrumentation-pyramid (setup.py): finished with status 'done'
  Created wheel for opentelemetry-instrumentation-pyramid: filename=opentelemetry_instrumentation_pyramid-0.24b0-py3-none-any.whl size=12304 sha256=9434900e7c787924d2551f782af3a50fccd9eaf0957e47682b517784260c6a94
  Stored in directory: /home/ocelotl/.cache/pip/wheels/2c/be/05/3d4fb2e3afa7509e2aaf369aa110282bd34f3dc3e5b4ba1964
Successfully built opentelemetry-instrumentation-pyramid
Installing collected packages: opentelemetry-instrumentation-pyramid
  Attempting uninstall: opentelemetry-instrumentation-pyramid
    Found existing installation: opentelemetry-instrumentation-pyramid 0.24b0
    Uninstalling opentelemetry-instrumentation-pyramid-0.24b0:
      Successfully uninstalled opentelemetry-instrumentation-pyramid-0.24b0
Successfully installed opentelemetry-instrumentation-pyramid-0.24b0
py39-test-instrumentation-pyramid run-test: commands[0] | pytest
============================= test session starts ==============================
platform linux -- Python 3.9.5, pytest-6.2.5, py-1.10.0, pluggy-1.0.0 -- /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/bin/python
cachedir: .tox/py39-test-instrumentation-pyramid/.pytest_cache
benchmark: 3.4.1 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)
rootdir: /home/ocelotl/github/ocelotl/opentelemetry-python-contrib, configfile: pytest.ini
plugins: benchmark-3.4.1
collecting ... collected 11 items

test_automatic.py::TestAutomatic::test_tween_list FAILED                 [  9%]
test_automatic.py::TestAutomatic::test_uninstrument 
-------------------------------- live log call ---------------------------------
WARNING  opentelemetry.instrumentation.instrumentor:instrumentor.py:98 Attempting to instrument while already instrumented
FAILED                                                                   [ 18%]
test_programmatic.py::TestProgrammatic::test_404 FAILED                  [ 27%]
test_programmatic.py::TestProgrammatic::test_exclude_lists FAILED        [ 36%]
test_programmatic.py::TestProgrammatic::test_internal_error FAILED       [ 45%]
test_programmatic.py::TestProgrammatic::test_not_recording FAILED        [ 54%]
test_programmatic.py::TestProgrammatic::test_response_headers FAILED     [ 63%]
test_programmatic.py::TestProgrammatic::test_simple FAILED               [ 72%]
test_programmatic.py::TestProgrammatic::test_tween_list FAILED           [ 81%]
test_programmatic.py::TestProgrammatic::test_uninstrument FAILED         [ 90%]
test_programmatic.py::TestProgrammatic::test_warnings FAILED             [100%]

=================================== FAILURES ===================================
________________________ TestAutomatic.test_tween_list _________________________

self = <pyramid.path.DottedNameResolver object at 0x7f1d981541f0>
value = 'opentelemetry.instrumentation.pyramid.trace_tween_factory'
package = <module 'opentelemetry.instrumentation.pyramid' from '/home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/__init__.py'>

    def _zope_dottedname_style(self, value, package):
        """ package.module.attr style """
        module = getattr(package, '__name__', None)  # package may be None
        if not module:
            module = None
        if value == '.':
            if module is None:
                raise ValueError(
                    'relative name %r irresolveable without package' % (value,)
                )
            name = module.split('.')
        else:
            name = value.split('.')
            if not name[0]:
                if module is None:
                    raise ValueError(
                        'relative name %r irresolveable without '
                        'package' % (value,)
                    )
                module = module.split('.')
                name.pop(0)
                while not name[0]:
                    module.pop()
                    name.pop(0)
                name = module + name
    
        used = name.pop(0)
        found = __import__(used)
        for n in name:
            used += '.' + n
            try:
>               found = getattr(found, n)
E               AttributeError: module 'opentelemetry.instrumentation.pyramid' has no attribute 'trace_tween_factory'

../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/path.py:387: AttributeError

During handling of the above exception, another exception occurred:

self = <tests.test_automatic.TestAutomatic testMethod=test_tween_list>

    def setUp(self):
        super().setUp()
    
        PyramidInstrumentor().instrument()
    
>       self.config = Configurator()

test_automatic.py:31: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/__init__.py:127: in _traced_init
    instance.include("opentelemetry.instrumentation.pyramid.callbacks")
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/__init__.py:666: in include
    c(configurator)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/callbacks.py:52: in includeme
    _insert_tween(config)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/callbacks.py:62: in _insert_tween
    config.add_tween(TWEEN_NAME, over=EXCVIEW)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/tweens.py:96: in add_tween
    return self._add_tween(
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/actions.py:572: in wrapper
    result = wrapped(self, *arg, **kw)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/tweens.py:118: in _add_tween
    tween_factory = self.maybe_dotted(tween_factory)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/__init__.py:740: in maybe_dotted
    return self.name_resolver.maybe_resolve(dotted)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/path.py:327: in maybe_resolve
    return self._resolve(dotted, package)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/path.py:334: in _resolve
    return self._zope_dottedname_style(dotted, package)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <pyramid.path.DottedNameResolver object at 0x7f1d981541f0>
value = 'opentelemetry.instrumentation.pyramid.trace_tween_factory'
package = <module 'opentelemetry.instrumentation.pyramid' from '/home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/__init__.py'>

    def _zope_dottedname_style(self, value, package):
        """ package.module.attr style """
        module = getattr(package, '__name__', None)  # package may be None
        if not module:
            module = None
        if value == '.':
            if module is None:
                raise ValueError(
                    'relative name %r irresolveable without package' % (value,)
                )
            name = module.split('.')
        else:
            name = value.split('.')
            if not name[0]:
                if module is None:
                    raise ValueError(
                        'relative name %r irresolveable without '
                        'package' % (value,)
                    )
                module = module.split('.')
                name.pop(0)
                while not name[0]:
                    module.pop()
                    name.pop(0)
                name = module + name
    
        used = name.pop(0)
        found = __import__(used)
        for n in name:
            used += '.' + n
            try:
                found = getattr(found, n)
            except AttributeError:
>               __import__(used)
E               ModuleNotFoundError: No module named 'opentelemetry.instrumentation.pyramid.trace_tween_factory'

../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/path.py:389: ModuleNotFoundError
_______________________ TestAutomatic.test_uninstrument ________________________

self = <pyramid.path.DottedNameResolver object at 0x7f1d98060be0>
value = 'opentelemetry.instrumentation.pyramid.trace_tween_factory'
package = <module 'opentelemetry.instrumentation.pyramid' from '/home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/__init__.py'>

    def _zope_dottedname_style(self, value, package):
        """ package.module.attr style """
        module = getattr(package, '__name__', None)  # package may be None
        if not module:
            module = None
        if value == '.':
            if module is None:
                raise ValueError(
                    'relative name %r irresolveable without package' % (value,)
                )
            name = module.split('.')
        else:
            name = value.split('.')
            if not name[0]:
                if module is None:
                    raise ValueError(
                        'relative name %r irresolveable without '
                        'package' % (value,)
                    )
                module = module.split('.')
                name.pop(0)
                while not name[0]:
                    module.pop()
                    name.pop(0)
                name = module + name
    
        used = name.pop(0)
        found = __import__(used)
        for n in name:
            used += '.' + n
            try:
>               found = getattr(found, n)
E               AttributeError: module 'opentelemetry.instrumentation.pyramid' has no attribute 'trace_tween_factory'

../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/path.py:387: AttributeError

During handling of the above exception, another exception occurred:

self = <tests.test_automatic.TestAutomatic testMethod=test_uninstrument>

    def setUp(self):
        super().setUp()
    
        PyramidInstrumentor().instrument()
    
>       self.config = Configurator()

test_automatic.py:31: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/__init__.py:127: in _traced_init
    instance.include("opentelemetry.instrumentation.pyramid.callbacks")
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/__init__.py:666: in include
    c(configurator)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/callbacks.py:52: in includeme
    _insert_tween(config)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/callbacks.py:62: in _insert_tween
    config.add_tween(TWEEN_NAME, over=EXCVIEW)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/tweens.py:96: in add_tween
    return self._add_tween(
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/actions.py:572: in wrapper
    result = wrapped(self, *arg, **kw)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/tweens.py:118: in _add_tween
    tween_factory = self.maybe_dotted(tween_factory)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/__init__.py:740: in maybe_dotted
    return self.name_resolver.maybe_resolve(dotted)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/path.py:327: in maybe_resolve
    return self._resolve(dotted, package)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/path.py:334: in _resolve
    return self._zope_dottedname_style(dotted, package)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <pyramid.path.DottedNameResolver object at 0x7f1d98060be0>
value = 'opentelemetry.instrumentation.pyramid.trace_tween_factory'
package = <module 'opentelemetry.instrumentation.pyramid' from '/home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/__init__.py'>

    def _zope_dottedname_style(self, value, package):
        """ package.module.attr style """
        module = getattr(package, '__name__', None)  # package may be None
        if not module:
            module = None
        if value == '.':
            if module is None:
                raise ValueError(
                    'relative name %r irresolveable without package' % (value,)
                )
            name = module.split('.')
        else:
            name = value.split('.')
            if not name[0]:
                if module is None:
                    raise ValueError(
                        'relative name %r irresolveable without '
                        'package' % (value,)
                    )
                module = module.split('.')
                name.pop(0)
                while not name[0]:
                    module.pop()
                    name.pop(0)
                name = module + name
    
        used = name.pop(0)
        found = __import__(used)
        for n in name:
            used += '.' + n
            try:
                found = getattr(found, n)
            except AttributeError:
>               __import__(used)
E               ModuleNotFoundError: No module named 'opentelemetry.instrumentation.pyramid.trace_tween_factory'

../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/path.py:389: ModuleNotFoundError
------------------------------ Captured log call -------------------------------
WARNING  opentelemetry.instrumentation.instrumentor:instrumentor.py:98 Attempting to instrument while already instrumented
__________________________ TestProgrammatic.test_404 ___________________________

self = <pyramid.path.DottedNameResolver object at 0x7f1d98000d30>
value = 'opentelemetry.instrumentation.pyramid.trace_tween_factory'
package = <module 'opentelemetry.instrumentation.pyramid' from '/home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/__init__.py'>

    def _zope_dottedname_style(self, value, package):
        """ package.module.attr style """
        module = getattr(package, '__name__', None)  # package may be None
        if not module:
            module = None
        if value == '.':
            if module is None:
                raise ValueError(
                    'relative name %r irresolveable without package' % (value,)
                )
            name = module.split('.')
        else:
            name = value.split('.')
            if not name[0]:
                if module is None:
                    raise ValueError(
                        'relative name %r irresolveable without '
                        'package' % (value,)
                    )
                module = module.split('.')
                name.pop(0)
                while not name[0]:
                    module.pop()
                    name.pop(0)
                name = module + name
    
        used = name.pop(0)
        found = __import__(used)
        for n in name:
            used += '.' + n
            try:
>               found = getattr(found, n)
E               AttributeError: module 'opentelemetry.instrumentation.pyramid' has no attribute 'trace_tween_factory'

../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/path.py:387: AttributeError

During handling of the above exception, another exception occurred:

self = <tests.test_programmatic.TestProgrammatic testMethod=test_404>

    def setUp(self):
        super().setUp()
>       config = Configurator()

test_programmatic.py:54: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/__init__.py:127: in _traced_init
    instance.include("opentelemetry.instrumentation.pyramid.callbacks")
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/__init__.py:666: in include
    c(configurator)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/callbacks.py:52: in includeme
    _insert_tween(config)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/callbacks.py:62: in _insert_tween
    config.add_tween(TWEEN_NAME, over=EXCVIEW)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/tweens.py:96: in add_tween
    return self._add_tween(
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/actions.py:572: in wrapper
    result = wrapped(self, *arg, **kw)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/tweens.py:118: in _add_tween
    tween_factory = self.maybe_dotted(tween_factory)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/__init__.py:740: in maybe_dotted
    return self.name_resolver.maybe_resolve(dotted)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/path.py:327: in maybe_resolve
    return self._resolve(dotted, package)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/path.py:334: in _resolve
    return self._zope_dottedname_style(dotted, package)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <pyramid.path.DottedNameResolver object at 0x7f1d98000d30>
value = 'opentelemetry.instrumentation.pyramid.trace_tween_factory'
package = <module 'opentelemetry.instrumentation.pyramid' from '/home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/__init__.py'>

    def _zope_dottedname_style(self, value, package):
        """ package.module.attr style """
        module = getattr(package, '__name__', None)  # package may be None
        if not module:
            module = None
        if value == '.':
            if module is None:
                raise ValueError(
                    'relative name %r irresolveable without package' % (value,)
                )
            name = module.split('.')
        else:
            name = value.split('.')
            if not name[0]:
                if module is None:
                    raise ValueError(
                        'relative name %r irresolveable without '
                        'package' % (value,)
                    )
                module = module.split('.')
                name.pop(0)
                while not name[0]:
                    module.pop()
                    name.pop(0)
                name = module + name
    
        used = name.pop(0)
        found = __import__(used)
        for n in name:
            used += '.' + n
            try:
                found = getattr(found, n)
            except AttributeError:
>               __import__(used)
E               ModuleNotFoundError: No module named 'opentelemetry.instrumentation.pyramid.trace_tween_factory'

../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/path.py:389: ModuleNotFoundError
_____________________ TestProgrammatic.test_exclude_lists ______________________

self = <pyramid.path.DottedNameResolver object at 0x7f1d97f91940>
value = 'opentelemetry.instrumentation.pyramid.trace_tween_factory'
package = <module 'opentelemetry.instrumentation.pyramid' from '/home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/__init__.py'>

    def _zope_dottedname_style(self, value, package):
        """ package.module.attr style """
        module = getattr(package, '__name__', None)  # package may be None
        if not module:
            module = None
        if value == '.':
            if module is None:
                raise ValueError(
                    'relative name %r irresolveable without package' % (value,)
                )
            name = module.split('.')
        else:
            name = value.split('.')
            if not name[0]:
                if module is None:
                    raise ValueError(
                        'relative name %r irresolveable without '
                        'package' % (value,)
                    )
                module = module.split('.')
                name.pop(0)
                while not name[0]:
                    module.pop()
                    name.pop(0)
                name = module + name
    
        used = name.pop(0)
        found = __import__(used)
        for n in name:
            used += '.' + n
            try:
>               found = getattr(found, n)
E               AttributeError: module 'opentelemetry.instrumentation.pyramid' has no attribute 'trace_tween_factory'

../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/path.py:387: AttributeError

During handling of the above exception, another exception occurred:

self = <tests.test_programmatic.TestProgrammatic testMethod=test_exclude_lists>

    def setUp(self):
        super().setUp()
>       config = Configurator()

test_programmatic.py:54: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/__init__.py:127: in _traced_init
    instance.include("opentelemetry.instrumentation.pyramid.callbacks")
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/__init__.py:666: in include
    c(configurator)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/callbacks.py:52: in includeme
    _insert_tween(config)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/callbacks.py:62: in _insert_tween
    config.add_tween(TWEEN_NAME, over=EXCVIEW)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/tweens.py:96: in add_tween
    return self._add_tween(
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/actions.py:572: in wrapper
    result = wrapped(self, *arg, **kw)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/tweens.py:118: in _add_tween
    tween_factory = self.maybe_dotted(tween_factory)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/__init__.py:740: in maybe_dotted
    return self.name_resolver.maybe_resolve(dotted)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/path.py:327: in maybe_resolve
    return self._resolve(dotted, package)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/path.py:334: in _resolve
    return self._zope_dottedname_style(dotted, package)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <pyramid.path.DottedNameResolver object at 0x7f1d97f91940>
value = 'opentelemetry.instrumentation.pyramid.trace_tween_factory'
package = <module 'opentelemetry.instrumentation.pyramid' from '/home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/__init__.py'>

    def _zope_dottedname_style(self, value, package):
        """ package.module.attr style """
        module = getattr(package, '__name__', None)  # package may be None
        if not module:
            module = None
        if value == '.':
            if module is None:
                raise ValueError(
                    'relative name %r irresolveable without package' % (value,)
                )
            name = module.split('.')
        else:
            name = value.split('.')
            if not name[0]:
                if module is None:
                    raise ValueError(
                        'relative name %r irresolveable without '
                        'package' % (value,)
                    )
                module = module.split('.')
                name.pop(0)
                while not name[0]:
                    module.pop()
                    name.pop(0)
                name = module + name
    
        used = name.pop(0)
        found = __import__(used)
        for n in name:
            used += '.' + n
            try:
                found = getattr(found, n)
            except AttributeError:
>               __import__(used)
E               ModuleNotFoundError: No module named 'opentelemetry.instrumentation.pyramid.trace_tween_factory'

../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/path.py:389: ModuleNotFoundError
_____________________ TestProgrammatic.test_internal_error _____________________

self = <pyramid.path.DottedNameResolver object at 0x7f1d97fd64f0>
value = 'opentelemetry.instrumentation.pyramid.trace_tween_factory'
package = <module 'opentelemetry.instrumentation.pyramid' from '/home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/__init__.py'>

    def _zope_dottedname_style(self, value, package):
        """ package.module.attr style """
        module = getattr(package, '__name__', None)  # package may be None
        if not module:
            module = None
        if value == '.':
            if module is None:
                raise ValueError(
                    'relative name %r irresolveable without package' % (value,)
                )
            name = module.split('.')
        else:
            name = value.split('.')
            if not name[0]:
                if module is None:
                    raise ValueError(
                        'relative name %r irresolveable without '
                        'package' % (value,)
                    )
                module = module.split('.')
                name.pop(0)
                while not name[0]:
                    module.pop()
                    name.pop(0)
                name = module + name
    
        used = name.pop(0)
        found = __import__(used)
        for n in name:
            used += '.' + n
            try:
>               found = getattr(found, n)
E               AttributeError: module 'opentelemetry.instrumentation.pyramid' has no attribute 'trace_tween_factory'

../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/path.py:387: AttributeError

During handling of the above exception, another exception occurred:

self = <tests.test_programmatic.TestProgrammatic testMethod=test_internal_error>

    def setUp(self):
        super().setUp()
>       config = Configurator()

test_programmatic.py:54: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/__init__.py:127: in _traced_init
    instance.include("opentelemetry.instrumentation.pyramid.callbacks")
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/__init__.py:666: in include
    c(configurator)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/callbacks.py:52: in includeme
    _insert_tween(config)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/callbacks.py:62: in _insert_tween
    config.add_tween(TWEEN_NAME, over=EXCVIEW)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/tweens.py:96: in add_tween
    return self._add_tween(
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/actions.py:572: in wrapper
    result = wrapped(self, *arg, **kw)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/tweens.py:118: in _add_tween
    tween_factory = self.maybe_dotted(tween_factory)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/__init__.py:740: in maybe_dotted
    return self.name_resolver.maybe_resolve(dotted)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/path.py:327: in maybe_resolve
    return self._resolve(dotted, package)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/path.py:334: in _resolve
    return self._zope_dottedname_style(dotted, package)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <pyramid.path.DottedNameResolver object at 0x7f1d97fd64f0>
value = 'opentelemetry.instrumentation.pyramid.trace_tween_factory'
package = <module 'opentelemetry.instrumentation.pyramid' from '/home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/__init__.py'>

    def _zope_dottedname_style(self, value, package):
        """ package.module.attr style """
        module = getattr(package, '__name__', None)  # package may be None
        if not module:
            module = None
        if value == '.':
            if module is None:
                raise ValueError(
                    'relative name %r irresolveable without package' % (value,)
                )
            name = module.split('.')
        else:
            name = value.split('.')
            if not name[0]:
                if module is None:
                    raise ValueError(
                        'relative name %r irresolveable without '
                        'package' % (value,)
                    )
                module = module.split('.')
                name.pop(0)
                while not name[0]:
                    module.pop()
                    name.pop(0)
                name = module + name
    
        used = name.pop(0)
        found = __import__(used)
        for n in name:
            used += '.' + n
            try:
                found = getattr(found, n)
            except AttributeError:
>               __import__(used)
E               ModuleNotFoundError: No module named 'opentelemetry.instrumentation.pyramid.trace_tween_factory'

../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/path.py:389: ModuleNotFoundError
_____________________ TestProgrammatic.test_not_recording ______________________

self = <pyramid.path.DottedNameResolver object at 0x7f1d980ba940>
value = 'opentelemetry.instrumentation.pyramid.trace_tween_factory'
package = <module 'opentelemetry.instrumentation.pyramid' from '/home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/__init__.py'>

    def _zope_dottedname_style(self, value, package):
        """ package.module.attr style """
        module = getattr(package, '__name__', None)  # package may be None
        if not module:
            module = None
        if value == '.':
            if module is None:
                raise ValueError(
                    'relative name %r irresolveable without package' % (value,)
                )
            name = module.split('.')
        else:
            name = value.split('.')
            if not name[0]:
                if module is None:
                    raise ValueError(
                        'relative name %r irresolveable without '
                        'package' % (value,)
                    )
                module = module.split('.')
                name.pop(0)
                while not name[0]:
                    module.pop()
                    name.pop(0)
                name = module + name
    
        used = name.pop(0)
        found = __import__(used)
        for n in name:
            used += '.' + n
            try:
>               found = getattr(found, n)
E               AttributeError: module 'opentelemetry.instrumentation.pyramid' has no attribute 'trace_tween_factory'

../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/path.py:387: AttributeError

During handling of the above exception, another exception occurred:

self = <tests.test_programmatic.TestProgrammatic testMethod=test_not_recording>

    def setUp(self):
        super().setUp()
>       config = Configurator()

test_programmatic.py:54: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/__init__.py:127: in _traced_init
    instance.include("opentelemetry.instrumentation.pyramid.callbacks")
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/__init__.py:666: in include
    c(configurator)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/callbacks.py:52: in includeme
    _insert_tween(config)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/callbacks.py:62: in _insert_tween
    config.add_tween(TWEEN_NAME, over=EXCVIEW)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/tweens.py:96: in add_tween
    return self._add_tween(
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/actions.py:572: in wrapper
    result = wrapped(self, *arg, **kw)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/tweens.py:118: in _add_tween
    tween_factory = self.maybe_dotted(tween_factory)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/__init__.py:740: in maybe_dotted
    return self.name_resolver.maybe_resolve(dotted)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/path.py:327: in maybe_resolve
    return self._resolve(dotted, package)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/path.py:334: in _resolve
    return self._zope_dottedname_style(dotted, package)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <pyramid.path.DottedNameResolver object at 0x7f1d980ba940>
value = 'opentelemetry.instrumentation.pyramid.trace_tween_factory'
package = <module 'opentelemetry.instrumentation.pyramid' from '/home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/__init__.py'>

    def _zope_dottedname_style(self, value, package):
        """ package.module.attr style """
        module = getattr(package, '__name__', None)  # package may be None
        if not module:
            module = None
        if value == '.':
            if module is None:
                raise ValueError(
                    'relative name %r irresolveable without package' % (value,)
                )
            name = module.split('.')
        else:
            name = value.split('.')
            if not name[0]:
                if module is None:
                    raise ValueError(
                        'relative name %r irresolveable without '
                        'package' % (value,)
                    )
                module = module.split('.')
                name.pop(0)
                while not name[0]:
                    module.pop()
                    name.pop(0)
                name = module + name
    
        used = name.pop(0)
        found = __import__(used)
        for n in name:
            used += '.' + n
            try:
                found = getattr(found, n)
            except AttributeError:
>               __import__(used)
E               ModuleNotFoundError: No module named 'opentelemetry.instrumentation.pyramid.trace_tween_factory'

../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/path.py:389: ModuleNotFoundError
____________________ TestProgrammatic.test_response_headers ____________________

self = <pyramid.path.DottedNameResolver object at 0x7f1d98174fd0>
value = 'opentelemetry.instrumentation.pyramid.trace_tween_factory'
package = <module 'opentelemetry.instrumentation.pyramid' from '/home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/__init__.py'>

    def _zope_dottedname_style(self, value, package):
        """ package.module.attr style """
        module = getattr(package, '__name__', None)  # package may be None
        if not module:
            module = None
        if value == '.':
            if module is None:
                raise ValueError(
                    'relative name %r irresolveable without package' % (value,)
                )
            name = module.split('.')
        else:
            name = value.split('.')
            if not name[0]:
                if module is None:
                    raise ValueError(
                        'relative name %r irresolveable without '
                        'package' % (value,)
                    )
                module = module.split('.')
                name.pop(0)
                while not name[0]:
                    module.pop()
                    name.pop(0)
                name = module + name
    
        used = name.pop(0)
        found = __import__(used)
        for n in name:
            used += '.' + n
            try:
>               found = getattr(found, n)
E               AttributeError: module 'opentelemetry.instrumentation.pyramid' has no attribute 'trace_tween_factory'

../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/path.py:387: AttributeError

During handling of the above exception, another exception occurred:

self = <tests.test_programmatic.TestProgrammatic testMethod=test_response_headers>

    def setUp(self):
        super().setUp()
>       config = Configurator()

test_programmatic.py:54: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/__init__.py:127: in _traced_init
    instance.include("opentelemetry.instrumentation.pyramid.callbacks")
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/__init__.py:666: in include
    c(configurator)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/callbacks.py:52: in includeme
    _insert_tween(config)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/callbacks.py:62: in _insert_tween
    config.add_tween(TWEEN_NAME, over=EXCVIEW)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/tweens.py:96: in add_tween
    return self._add_tween(
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/actions.py:572: in wrapper
    result = wrapped(self, *arg, **kw)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/tweens.py:118: in _add_tween
    tween_factory = self.maybe_dotted(tween_factory)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/__init__.py:740: in maybe_dotted
    return self.name_resolver.maybe_resolve(dotted)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/path.py:327: in maybe_resolve
    return self._resolve(dotted, package)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/path.py:334: in _resolve
    return self._zope_dottedname_style(dotted, package)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <pyramid.path.DottedNameResolver object at 0x7f1d98174fd0>
value = 'opentelemetry.instrumentation.pyramid.trace_tween_factory'
package = <module 'opentelemetry.instrumentation.pyramid' from '/home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/__init__.py'>

    def _zope_dottedname_style(self, value, package):
        """ package.module.attr style """
        module = getattr(package, '__name__', None)  # package may be None
        if not module:
            module = None
        if value == '.':
            if module is None:
                raise ValueError(
                    'relative name %r irresolveable without package' % (value,)
                )
            name = module.split('.')
        else:
            name = value.split('.')
            if not name[0]:
                if module is None:
                    raise ValueError(
                        'relative name %r irresolveable without '
                        'package' % (value,)
                    )
                module = module.split('.')
                name.pop(0)
                while not name[0]:
                    module.pop()
                    name.pop(0)
                name = module + name
    
        used = name.pop(0)
        found = __import__(used)
        for n in name:
            used += '.' + n
            try:
                found = getattr(found, n)
            except AttributeError:
>               __import__(used)
E               ModuleNotFoundError: No module named 'opentelemetry.instrumentation.pyramid.trace_tween_factory'

../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/path.py:389: ModuleNotFoundError
_________________________ TestProgrammatic.test_simple _________________________

self = <pyramid.path.DottedNameResolver object at 0x7f1d9803e940>
value = 'opentelemetry.instrumentation.pyramid.trace_tween_factory'
package = <module 'opentelemetry.instrumentation.pyramid' from '/home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/__init__.py'>

    def _zope_dottedname_style(self, value, package):
        """ package.module.attr style """
        module = getattr(package, '__name__', None)  # package may be None
        if not module:
            module = None
        if value == '.':
            if module is None:
                raise ValueError(
                    'relative name %r irresolveable without package' % (value,)
                )
            name = module.split('.')
        else:
            name = value.split('.')
            if not name[0]:
                if module is None:
                    raise ValueError(
                        'relative name %r irresolveable without '
                        'package' % (value,)
                    )
                module = module.split('.')
                name.pop(0)
                while not name[0]:
                    module.pop()
                    name.pop(0)
                name = module + name
    
        used = name.pop(0)
        found = __import__(used)
        for n in name:
            used += '.' + n
            try:
>               found = getattr(found, n)
E               AttributeError: module 'opentelemetry.instrumentation.pyramid' has no attribute 'trace_tween_factory'

../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/path.py:387: AttributeError

During handling of the above exception, another exception occurred:

self = <tests.test_programmatic.TestProgrammatic testMethod=test_simple>

    def setUp(self):
        super().setUp()
>       config = Configurator()

test_programmatic.py:54: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/__init__.py:127: in _traced_init
    instance.include("opentelemetry.instrumentation.pyramid.callbacks")
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/__init__.py:666: in include
    c(configurator)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/callbacks.py:52: in includeme
    _insert_tween(config)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/callbacks.py:62: in _insert_tween
    config.add_tween(TWEEN_NAME, over=EXCVIEW)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/tweens.py:96: in add_tween
    return self._add_tween(
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/actions.py:572: in wrapper
    result = wrapped(self, *arg, **kw)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/tweens.py:118: in _add_tween
    tween_factory = self.maybe_dotted(tween_factory)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/__init__.py:740: in maybe_dotted
    return self.name_resolver.maybe_resolve(dotted)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/path.py:327: in maybe_resolve
    return self._resolve(dotted, package)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/path.py:334: in _resolve
    return self._zope_dottedname_style(dotted, package)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <pyramid.path.DottedNameResolver object at 0x7f1d9803e940>
value = 'opentelemetry.instrumentation.pyramid.trace_tween_factory'
package = <module 'opentelemetry.instrumentation.pyramid' from '/home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/__init__.py'>

    def _zope_dottedname_style(self, value, package):
        """ package.module.attr style """
        module = getattr(package, '__name__', None)  # package may be None
        if not module:
            module = None
        if value == '.':
            if module is None:
                raise ValueError(
                    'relative name %r irresolveable without package' % (value,)
                )
            name = module.split('.')
        else:
            name = value.split('.')
            if not name[0]:
                if module is None:
                    raise ValueError(
                        'relative name %r irresolveable without '
                        'package' % (value,)
                    )
                module = module.split('.')
                name.pop(0)
                while not name[0]:
                    module.pop()
                    name.pop(0)
                name = module + name
    
        used = name.pop(0)
        found = __import__(used)
        for n in name:
            used += '.' + n
            try:
                found = getattr(found, n)
            except AttributeError:
>               __import__(used)
E               ModuleNotFoundError: No module named 'opentelemetry.instrumentation.pyramid.trace_tween_factory'

../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/path.py:389: ModuleNotFoundError
_______________________ TestProgrammatic.test_tween_list _______________________

self = <pyramid.path.DottedNameResolver object at 0x7f1d980baf40>
value = 'opentelemetry.instrumentation.pyramid.trace_tween_factory'
package = <module 'opentelemetry.instrumentation.pyramid' from '/home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/__init__.py'>

    def _zope_dottedname_style(self, value, package):
        """ package.module.attr style """
        module = getattr(package, '__name__', None)  # package may be None
        if not module:
            module = None
        if value == '.':
            if module is None:
                raise ValueError(
                    'relative name %r irresolveable without package' % (value,)
                )
            name = module.split('.')
        else:
            name = value.split('.')
            if not name[0]:
                if module is None:
                    raise ValueError(
                        'relative name %r irresolveable without '
                        'package' % (value,)
                    )
                module = module.split('.')
                name.pop(0)
                while not name[0]:
                    module.pop()
                    name.pop(0)
                name = module + name
    
        used = name.pop(0)
        found = __import__(used)
        for n in name:
            used += '.' + n
            try:
>               found = getattr(found, n)
E               AttributeError: module 'opentelemetry.instrumentation.pyramid' has no attribute 'trace_tween_factory'

../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/path.py:387: AttributeError

During handling of the above exception, another exception occurred:

self = <tests.test_programmatic.TestProgrammatic testMethod=test_tween_list>

    def setUp(self):
        super().setUp()
>       config = Configurator()

test_programmatic.py:54: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/__init__.py:127: in _traced_init
    instance.include("opentelemetry.instrumentation.pyramid.callbacks")
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/__init__.py:666: in include
    c(configurator)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/callbacks.py:52: in includeme
    _insert_tween(config)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/callbacks.py:62: in _insert_tween
    config.add_tween(TWEEN_NAME, over=EXCVIEW)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/tweens.py:96: in add_tween
    return self._add_tween(
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/actions.py:572: in wrapper
    result = wrapped(self, *arg, **kw)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/tweens.py:118: in _add_tween
    tween_factory = self.maybe_dotted(tween_factory)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/__init__.py:740: in maybe_dotted
    return self.name_resolver.maybe_resolve(dotted)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/path.py:327: in maybe_resolve
    return self._resolve(dotted, package)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/path.py:334: in _resolve
    return self._zope_dottedname_style(dotted, package)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <pyramid.path.DottedNameResolver object at 0x7f1d980baf40>
value = 'opentelemetry.instrumentation.pyramid.trace_tween_factory'
package = <module 'opentelemetry.instrumentation.pyramid' from '/home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/__init__.py'>

    def _zope_dottedname_style(self, value, package):
        """ package.module.attr style """
        module = getattr(package, '__name__', None)  # package may be None
        if not module:
            module = None
        if value == '.':
            if module is None:
                raise ValueError(
                    'relative name %r irresolveable without package' % (value,)
                )
            name = module.split('.')
        else:
            name = value.split('.')
            if not name[0]:
                if module is None:
                    raise ValueError(
                        'relative name %r irresolveable without '
                        'package' % (value,)
                    )
                module = module.split('.')
                name.pop(0)
                while not name[0]:
                    module.pop()
                    name.pop(0)
                name = module + name
    
        used = name.pop(0)
        found = __import__(used)
        for n in name:
            used += '.' + n
            try:
                found = getattr(found, n)
            except AttributeError:
>               __import__(used)
E               ModuleNotFoundError: No module named 'opentelemetry.instrumentation.pyramid.trace_tween_factory'

../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/path.py:389: ModuleNotFoundError
______________________ TestProgrammatic.test_uninstrument ______________________

self = <pyramid.path.DottedNameResolver object at 0x7f1d98157a90>
value = 'opentelemetry.instrumentation.pyramid.trace_tween_factory'
package = <module 'opentelemetry.instrumentation.pyramid' from '/home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/__init__.py'>

    def _zope_dottedname_style(self, value, package):
        """ package.module.attr style """
        module = getattr(package, '__name__', None)  # package may be None
        if not module:
            module = None
        if value == '.':
            if module is None:
                raise ValueError(
                    'relative name %r irresolveable without package' % (value,)
                )
            name = module.split('.')
        else:
            name = value.split('.')
            if not name[0]:
                if module is None:
                    raise ValueError(
                        'relative name %r irresolveable without '
                        'package' % (value,)
                    )
                module = module.split('.')
                name.pop(0)
                while not name[0]:
                    module.pop()
                    name.pop(0)
                name = module + name
    
        used = name.pop(0)
        found = __import__(used)
        for n in name:
            used += '.' + n
            try:
>               found = getattr(found, n)
E               AttributeError: module 'opentelemetry.instrumentation.pyramid' has no attribute 'trace_tween_factory'

../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/path.py:387: AttributeError

During handling of the above exception, another exception occurred:

self = <tests.test_programmatic.TestProgrammatic testMethod=test_uninstrument>

    def setUp(self):
        super().setUp()
>       config = Configurator()

test_programmatic.py:54: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/__init__.py:127: in _traced_init
    instance.include("opentelemetry.instrumentation.pyramid.callbacks")
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/__init__.py:666: in include
    c(configurator)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/callbacks.py:52: in includeme
    _insert_tween(config)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/callbacks.py:62: in _insert_tween
    config.add_tween(TWEEN_NAME, over=EXCVIEW)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/tweens.py:96: in add_tween
    return self._add_tween(
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/actions.py:572: in wrapper
    result = wrapped(self, *arg, **kw)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/tweens.py:118: in _add_tween
    tween_factory = self.maybe_dotted(tween_factory)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/__init__.py:740: in maybe_dotted
    return self.name_resolver.maybe_resolve(dotted)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/path.py:327: in maybe_resolve
    return self._resolve(dotted, package)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/path.py:334: in _resolve
    return self._zope_dottedname_style(dotted, package)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <pyramid.path.DottedNameResolver object at 0x7f1d98157a90>
value = 'opentelemetry.instrumentation.pyramid.trace_tween_factory'
package = <module 'opentelemetry.instrumentation.pyramid' from '/home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/__init__.py'>

    def _zope_dottedname_style(self, value, package):
        """ package.module.attr style """
        module = getattr(package, '__name__', None)  # package may be None
        if not module:
            module = None
        if value == '.':
            if module is None:
                raise ValueError(
                    'relative name %r irresolveable without package' % (value,)
                )
            name = module.split('.')
        else:
            name = value.split('.')
            if not name[0]:
                if module is None:
                    raise ValueError(
                        'relative name %r irresolveable without '
                        'package' % (value,)
                    )
                module = module.split('.')
                name.pop(0)
                while not name[0]:
                    module.pop()
                    name.pop(0)
                name = module + name
    
        used = name.pop(0)
        found = __import__(used)
        for n in name:
            used += '.' + n
            try:
                found = getattr(found, n)
            except AttributeError:
>               __import__(used)
E               ModuleNotFoundError: No module named 'opentelemetry.instrumentation.pyramid.trace_tween_factory'

../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/path.py:389: ModuleNotFoundError
________________________ TestProgrammatic.test_warnings ________________________

self = <pyramid.path.DottedNameResolver object at 0x7f1d980923a0>
value = 'opentelemetry.instrumentation.pyramid.trace_tween_factory'
package = <module 'opentelemetry.instrumentation.pyramid' from '/home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/__init__.py'>

    def _zope_dottedname_style(self, value, package):
        """ package.module.attr style """
        module = getattr(package, '__name__', None)  # package may be None
        if not module:
            module = None
        if value == '.':
            if module is None:
                raise ValueError(
                    'relative name %r irresolveable without package' % (value,)
                )
            name = module.split('.')
        else:
            name = value.split('.')
            if not name[0]:
                if module is None:
                    raise ValueError(
                        'relative name %r irresolveable without '
                        'package' % (value,)
                    )
                module = module.split('.')
                name.pop(0)
                while not name[0]:
                    module.pop()
                    name.pop(0)
                name = module + name
    
        used = name.pop(0)
        found = __import__(used)
        for n in name:
            used += '.' + n
            try:
>               found = getattr(found, n)
E               AttributeError: module 'opentelemetry.instrumentation.pyramid' has no attribute 'trace_tween_factory'

../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/path.py:387: AttributeError

During handling of the above exception, another exception occurred:

self = <tests.test_programmatic.TestProgrammatic testMethod=test_warnings>

    def setUp(self):
        super().setUp()
>       config = Configurator()

test_programmatic.py:54: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/__init__.py:127: in _traced_init
    instance.include("opentelemetry.instrumentation.pyramid.callbacks")
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/__init__.py:666: in include
    c(configurator)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/callbacks.py:52: in includeme
    _insert_tween(config)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/callbacks.py:62: in _insert_tween
    config.add_tween(TWEEN_NAME, over=EXCVIEW)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/tweens.py:96: in add_tween
    return self._add_tween(
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/actions.py:572: in wrapper
    result = wrapped(self, *arg, **kw)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/tweens.py:118: in _add_tween
    tween_factory = self.maybe_dotted(tween_factory)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/config/__init__.py:740: in maybe_dotted
    return self.name_resolver.maybe_resolve(dotted)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/path.py:327: in maybe_resolve
    return self._resolve(dotted, package)
../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/path.py:334: in _resolve
    return self._zope_dottedname_style(dotted, package)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <pyramid.path.DottedNameResolver object at 0x7f1d980923a0>
value = 'opentelemetry.instrumentation.pyramid.trace_tween_factory'
package = <module 'opentelemetry.instrumentation.pyramid' from '/home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/opentelemetry/instrumentation/pyramid/__init__.py'>

    def _zope_dottedname_style(self, value, package):
        """ package.module.attr style """
        module = getattr(package, '__name__', None)  # package may be None
        if not module:
            module = None
        if value == '.':
            if module is None:
                raise ValueError(
                    'relative name %r irresolveable without package' % (value,)
                )
            name = module.split('.')
        else:
            name = value.split('.')
            if not name[0]:
                if module is None:
                    raise ValueError(
                        'relative name %r irresolveable without '
                        'package' % (value,)
                    )
                module = module.split('.')
                name.pop(0)
                while not name[0]:
                    module.pop()
                    name.pop(0)
                name = module + name
    
        used = name.pop(0)
        found = __import__(used)
        for n in name:
            used += '.' + n
            try:
                found = getattr(found, n)
            except AttributeError:
>               __import__(used)
E               ModuleNotFoundError: No module named 'opentelemetry.instrumentation.pyramid.trace_tween_factory'

../../../.tox/py39-test-instrumentation-pyramid/lib/python3.9/site-packages/pyramid/path.py:389: ModuleNotFoundError
============================== 11 failed in 0.68s ==============================
ERROR: InvocationError for command /home/ocelotl/github/ocelotl/opentelemetry-python-contrib/.tox/py39-test-instrumentation-pyramid/bin/pytest (exited with code 1)
___________________________________ summary ____________________________________
ERROR:   py39-test-instrumentation-pyramid: commands failed
```